cpp/CMakeLists.txt

```diff
@@ -262,11 +262,18 @@ if(ARROW_BUILD_TESTS OR ARROW_BUILD_BENCHMARKS)
   set(ARROW_WITH_ZLIB ON)
 endif()
 
-if(ARROW_BUILD_TESTS)
-  # JSON parsing of arrays is required for Arrow unit tests
+if((ARROW_BUILD_TESTS OR ARROW_BUILD_INTEGRATION) AND NOT ARROW_JSON)
+  message(FATAL_ERROR "JSON parsing of arrays is required for Arrow tests")
+endif()
+
+if(ARROW_FLIGHT)
```

Comment:
This changes removes the implicit activation of `ARROW_IPC=ON` when `ARROW_BUILD_TEST=ON`, is this warranted? (Should this read ARROW_FLIGHT OR ARROW_BUILD_TESTS)

Reply:
That is a good catch, I tried:
```
-DARROW_BUILD_TESTS=ON -DARROW_IPC=OFF
```
and many tests failed due to missing the symbols for `ArrayFromJSON` which is part of the IPC sources.  I'll update this shortly.